src/logic/equiv/basic.lean

```diff
@@ -1214,6 +1215,13 @@ calc α × (β ⊕ γ) ≃ (β ⊕ γ) × α       : prod_comm _ _
 @[simp] theorem prod_sum_distrib_apply_right {α β γ} (a : α) (c : γ) :
    prod_sum_distrib α β γ (a, sum.inr c) = sum.inr (a, c) := rfl
 
+/-- An indexed sum of disjoint sums of types is equivalent to the sum of the indexed sums. -/
+def sigma_sum_distrib {ι : Type*} (α β : ι → Type*) : (Σ i, α i ⊕ β i) ≃ (Σ i, α i) ⊕ Σ i, β i :=
```

Comment:
What does `@[simps]` generate if you add it here? You can use `simps?` to find out.

Reply:
An `apply` and a `symm_apply` lemma - seems worth it.

Reply:
They're both a bit of a mess, but I guess it's still better than what you get if you unfold `sigma_sum_distrib` directly.